api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPersonDaoImpl.java

```diff
@@ -45,33 +49,56 @@ public Person getPersonByUuid(String uuid) {
 	}
 	
 	@Override
-	public Collection<Person> findPersonsByName(String name) {
-		return personService.getPeople(name, false);
-	}
-	
-	@Override
-	@SuppressWarnings("unchecked")
-	public Collection<Person> findPersonsByBirthDate(Date birthDate) {
-		return sessionFactory.getCurrentSession().createCriteria(Person.class).add(eq("birthdate", birthDate)).list();
+	public List<PersonAttribute> getActiveAttributesByPersonAndAttributeTypeUuid(Person person,
+	        String personAttributeTypeUuid) {
+		return (List<PersonAttribute>) sessionFactory.getCurrentSession().createCriteria(PersonAttribute.class)
+		        .createAlias("person", "p", JoinType.INNER_JOIN, eq("p.id", person.getId()))
+		        .createAlias("attributeType", "pat").add(eq("pat.uuid", personAttributeTypeUuid)).add(eq("voided", false))
+		        .list();
 	}
 	
 	@Override
-	public Collection<Person> findSimilarPeople(String name, Integer birthYear, String gender) {
-		return personService.getSimilarPeople(name, birthYear, gender);
+	public Collection<Person> searchForPeople(StringOrListParam name, TokenOrListParam gender, DateRangeParam birthDate,
+	        StringOrListParam city, StringOrListParam state, StringOrListParam postalCode, StringOrListParam country,
+	        SortSpec sort) {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Person.class);
+		
+		handleNames(criteria, name, null, null);
+		handleGender("gender", gender).ifPresent(criteria::add);
+		handleDateRange("birthdate", birthDate).ifPresent(criteria::add);
+		handlePersonAddress("pad", city, state, postalCode, country).ifPresent(c -> {
+			criteria.createAlias("addresses", "pad");
+			criteria.add(c);
+		});
+		if (!containsAlias(criteria, "pn")) {
```

Comment:
Again... this is creating the alias regardless of what is being queried, which is not what we want to do. I might change your `paramToProp()` to something like this:

```java
protected String paramToProp(String param) {
		String prop = null;

		switch (param) {
			case "name":
				if (!containsAlias(criteria, "pn")) {
					criteria.createAlias("names", "pn");
				}

				return "pn.givenName";
			case "birthDate":
				return "birthdate";
			case "city":
				prop = "pad.cityVillage";
			case "state":
				prop = "pad.stateProvince";
			case "postalCode":
				prop = "pad.postalCode";
			case "country":
				prop = "pad.country";
                
                if (!containsAlias(criteria, "pad")) {
					criteria.createAlias("addresses", "pad");
				}
				
				return prop;
			default:
				return null;
		}
	}
```

Reply:
Yes, I'm working on this right now.